### ISSUE TITLE
No object detected with custom dataset!

### ISSUE THREAD

**Issue by YangHe712** [traveller59/second.pytorch#140]
Hello, @traveller59 
I create a dataset myself according to the format of KITTI (with all information available in addition to the 2D image bbox). The dataset contains 2800+ frames.

I do training based on my own dataset and it seems to work normally. But when I use the inference module provided by you, it completely detects nothing!

I only modified the class size info in car_fhd.config with the average size I get from my dataset. Also, I removed the two filters in the config file, and changed the anchor_ranges: [0, -40.0, -1.78, 70.4, 40.0, -1.78]  (In the old version the z center is -1.78 but in the neweset one it is changed to -1, I don't know the reason of your change so I changed it back to -1.78.  Because the last time I trained with only 80 frames of my dataset as a test, I could detect several cars and the z value I set was -1.78 then).

So, somehow I got lost where else I possibly made mistakes that lead to the complete fail detection. I really appreciate if you could give me some possible solultions.

The bev_map I got inferenced as follows:
![image](https://user-images.githubusercontent.com/33030588/55621033-bb46ba00-579c-11e9-973e-05c0e1cbe9d8.png)

**Comment by traveller59** [comment]
in v1.6.0alpha, I have changed all boxes from kitti format ([0.5, 0.5, 0], box center) to to CENTER format (0.5, 0.5, 0.5), so the z center in anchors need to be changed to center format. the ```post_center_limit_range``` also need to be changed.

Before you train your custom data, you should ensure the 3d boxes is correct in my web visualization tool.

**Comment by YangHe712** [comment]
> in v1.6.0alpha, I have changed all boxes from kitti format ([0.5, 0.5, 0], box center) to to CENTER format (0.5, 0.5, 0.5), so the z center in anchors need to be changed to center format. the `post_center_limit_range` also need to be changed.
> 
> Before you train your custom data, you should ensure the 3d boxes is correct in my web visualization tool.

@traveller59 thanks for your reply first!

1. I didn't change the `post_center_limit_range: [0, -40, -2.2, 70.4, 40, 0.8] `. The pc x,y range of my dataset is smaller than KITTI's, and I lift the ground to -1.78 as well as in KITTI, so I still adopt your point cloud range  in config file. So, if I keep your `post_center_limit_range:` unchanged, I should change z center to -1 instead of -1.78, right? or ?

2. I have visualized my dataset in your webviewer before, the labels for 3D bbox and rotation should be correct according to that.

So, what should I modify based on these?

thanks for much and have a nice weekend!

**Comment by traveller59** [comment]
you need to change anchor center to -1 (mean of z-center of gt boxes in your dataset) and retrain your dataset...
you can use pretrained models to inference your data first.

**Comment by cslxiao** [comment]
Using the pretrained model for customized data didn't work, although I have aligned the ground plane by adding a bias to z-value.

With kitti point cloud as input, I can get reasonable results. When I replace the input with point cloud collected with hesai pandar40p, I get the following prediction:

![screenshot](https://user-images.githubusercontent.com/1681010/83000286-b72ec900-a03c-11ea-8b85-60f978ac0c55.png)

Did I miss some preprocessing? Or it is just because the pretrained model cannot generalize to point cloud collected with different sensors?

**Comment by cslxiao** [comment]
sorry to bother! It is caused by the unnormalized intensity. After dividing the intensity by 255, the results get right.
